posts/2024-10-17-Rust-1.82.0.md

```diff
@@ -0,0 +1,320 @@
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.82.0. Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via `rustup`, you can get 1.82.0 with:
+
+```console
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.82.0](https://doc.rust-lang.org/nightly/releases.html#version-1820-2024-10-17).
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.82.0 stable
+
+### `cargo info`
+
+Cargo now has an [`info` subcommand](https://doc.rust-lang.org/nightly/cargo/commands/cargo-info.html) to display information about a package in the registry, fulfilling a [long standing request](https://github.com/rust-lang/cargo/issues/948) just shy of its tenth anniversary! Several third-party extensions like this have been written over the years, and this implementation was developed as [cargo-information](https://crates.io/crates/cargo-information) before merging into Cargo itself.
+
+For example, here's what you could see for `cargo info cc`:
+
+<pre>
+<b><span style="color:#0A0">cc</span></b> <b><span style="color:#0AA">#build-dependencies</span></b>
+A build-time dependency for Cargo build scripts to assist in invoking the native
+C compiler to compile native C code into a static archive to be linked into Rust
+code.
+<b><span style="color:#0A0">version:</span></b> 1.1.23 <b><span style="color:#A50">(latest 1.1.30)</span></b>
+<b><span style="color:#0A0">license:</span></b> MIT OR Apache-2.0
+<b><span style="color:#0A0">rust-version:</span></b> 1.63
+<b><span style="color:#0A0">documentation:</span></b> https://docs.rs/cc
+<b><span style="color:#0A0">homepage:</span></b> https://github.com/rust-lang/cc-rs
+<b><span style="color:#0A0">repository:</span></b> https://github.com/rust-lang/cc-rs
+<b><span style="color:#0A0">crates.io:</span></b> https://crates.io/crates/cc/1.1.23
+<b><span style="color:#0A0">features:</span></b>
+  jobserver = []
+  parallel  = [dep:libc, dep:jobserver]
+<b><span style="color:#0AA">note</span></b><b>:</b> to see how you depend on cc, run `<b><span style="color:#0AA">cargo tree --invert --package cc@1.1.23</span></b>`
+</pre>
+
+By default `cargo info` describes the package version in the local `Cargo.lock`, if any, but as you can see it will indicate when there's a newer version too, and `cargo info cc@1.1.30` would report on that. The feature list also indicates a `+` next to any that are enabled.
+
+### Apple target promotions
+
+#### macOS on ARM64 is now Tier 1
```

Comment:
```suggestion
#### macOS on AArch64 is now Tier 1
```

So, this is an awkward one. Pretty much everywhere *Rust* refers to this architecture, we[^1] call it "aarch64". *Apple* appears to use "arm64" in code paths and "Apple Silicon" in user-facing / marketing venues.

For the purpose of this, a Rust-specific blog post, I think we should primarily refer to the architecture as `aarch64` if it's code or "AArch64" if it's human talk (according to [the Wikipedia article](https://en.wikipedia.org/wiki/AArch64)). We should then extend the aliases section below to mention "ARM64".

[^1]: If I recall correctly, this was basically a unilateral decision on my part to better match our existing targets.

Reply:
On the other hand, our [platform support](https://doc.rust-lang.org/nightly/rustc/platform-support.html) page consistently uses "ARM64" in the target "notes" descriptions, and there are also a few in triples: T2 `arm64ec-pc-windows-msvc`, T3 `arm64e-apple-{darwin,ios,tvos}` and `arm64_32-apple-watchos`.

The [Rust 1.49.0](https://blog.rust-lang.org/2020/12/31/Rust-1.49.0.html) announcement spelled out "64-bit ARM" for both Linux and macOS -- would that be acceptable to you here?

Reply:
> spelled out "64-bit ARM"

That'd OK. At least we skip picking one name or the other.

> consistently uses "ARM64"

It may be relevant that Wikipedia redirects `ARM64` to the aforementioned `AArch64` page, and only uses "ARM64" once as an "also known as". Perhaps we should follow suit?

> and there are also a few in triples: T2 `arm64ec-pc-windows-msvc`, T3 `arm64e-apple-{darwin,ios,tvos}`

To my limited knowledge arm64**e** and arm64**ec** are ABIs, not processor architectures. It may be a sign they should not be present there in these target triple?

> and `arm64_32-apple-watchos`.

The same may be true for `arm64_32`?

Reply:
> > spelled out "64-bit ARM"
> 
> That'd be OK. At least we skip picking one name or the other.

At the very least, I think this is not time or place to litigate it. :) I'll make the change.